core/src/marker/markerManager.cpp

```diff
@@ -297,6 +298,27 @@ bool MarkerManager::buildStyling(Marker& marker) {
     try {
         // Update the draw rule for the marker.
         YAML::Node node = YAML::Load(marker.stylingString());
+
+        // Check if marker styling defines a layer in the yaml scene
+        if (Node layer = node["layer"]) {
```

Comment:
I agree on the "not" using of yaml here. Based on the initial requirements from @ecgreb it seems they wanted to use the same method for their purposes. (Refering initial discussion on slack).

I had suggested the exact same function method.